File: COSC 4315/assign2/Scala/src/RecursiveFactorial.scala
class RecursiveFactorial{

    def recursiveFactorial(num: BigInt): BigInt = {

        if(num.compareTo(0) == -1) return -1

        if (num.compareTo(0) == 0) return 1

        return num * recursiveFactorial(num - 1)
    }
}

File: COSC 4315/assign2/Kotlin/src/RecursiveFactorial.kt
import java.math.BigInteger

class RecursiveFactorial {

    fun recursiveFactorial(num: BigInteger): BigInteger{

        if(num.compareTo(BigInteger.valueOf(0)) == -1) return BigInteger.valueOf(-1)

        if (num.compareTo(BigInteger.valueOf(0)) == 0) return BigInteger.valueOf(1)

        return num.multiply(recursiveFactorial(num - BigInteger.valueOf(1)))
    }
}
